Tests for version 6.3 fail when run from inside egg/wheel

See https://github.com/zopefoundation/zopetoolkit/actions/runs/7828690859/job/21359091342?pr=74#step:6:572

I see this on OpenIndiana too.

For reference, the failure is this:
```
Failure in test test_xml_report_with_failures (zope.testrunner.tests.test_xml_output.TextXMLOutputWithErrors)
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/unittest/case.py", line 60, in testPartExecutor
    yield
  File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/unittest/case.py", line 676, in run
    self._callTestMethod(testMethod)
  File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/unittest/case.py", line 633, in _callTestMethod
    method()
  File "/home/runner/work/zopetoolkit/zopetoolkit/.tox/py38/eggs/zope.testrunner-6.3-py3.8.egg/zope/testrunner/tests/test_xml_output.py", line 171, in test_xml_report_with_failures
    self.assertTrue(report.exists())
  File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/unittest/case.py", line 765, in assertTrue
    raise self.failureException(msg)
AssertionError: False is not true
```

--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -5,6 +5,8 @@
 6.4 (unreleased)
 ================
 
+- Fix XML tests when running in distribution resp. separately.
+  (`#163 <https://github.com/zopefoundation/zope.testrunner/issues/163>`_)
 
 6.3 (2024-02-07)
 ================

--- a/src/zope/testrunner/tests/test_xml_output.py
+++ b/src/zope/testrunner/tests/test_xml_output.py
@@ -15,8 +15,6 @@
 """
 import contextlib
 import io
-import os
-import os.path
 import shutil
 import sys
 import tempfile
@@ -29,13 +27,12 @@ from zope import testrunner
 class Base(unittest.TestCase):
 
     def tearDown(self):
-        self._cleanup(self.reports_folder)
+        self._cleanup(self.tmpdir)
         return super().tearDown()
 
     def _cleanup(self, path):
-        if path.exists():
-            [x.unlink() for x in path.iterdir()]
-            path.rmdir()
+        assert path.exists()
+        shutil.rmtree(path)
 
 
 class TestXMLOutput(Base):
@@ -53,15 +50,14 @@ class TestXMLOutput(Base):
     """
 
     def setUp(self):
-        this_directory = os.path.split(__file__)[0]
-        directory_with_tests = Path(this_directory, 'testrunner-ex')
+        self.tmpdir = Path(tempfile.mkdtemp())
+        directory_with_tests = Path(Path(__file__).parent, 'testrunner-ex')
         self.arg_defaults = [
             '--path', str(directory_with_tests),
             '--tests-pattern', '^sampletestsf?$',
         ]
-        self.default_argv = 'test -u --xml=. -t sample3'.split()
-        self.reports_folder = Path('testreports')
-        self._cleanup(self.reports_folder)
+        self.default_argv = f'test -u --xml={self.tmpdir} -t sample3'.split()
+        self.reports_folder = self.tmpdir / 'testreports'
 
     def _run_tests(self):
         stream1 = io.StringIO()
@@ -109,26 +105,18 @@ class TextXMLOutputWithErrors(Base):
     """
 
     def setUp(self):
-        this_directory = os.path.split(__file__)[0]
-        tmpdir = tempfile.mkdtemp()
-        directory_with_tests = os.path.join(tmpdir, 'testrunner-ex')
-        self.reports_folder = Path('testreports')
+        self.tmpdir = Path(tempfile.mkdtemp())
+        directory_with_tests = self.tmpdir / 'testrunner-ex'
+        self.reports_folder = self.tmpdir / 'testreports'
 
-        source = os.path.join(this_directory, 'testrunner-ex')
-        n = len(source) + 1
-        for root, dirs, files in os.walk(source):
-            dirs[:] = [d for d in dirs if d != ".svn"]  # prune cruft
-            os.mkdir(os.path.join(directory_with_tests, root[n:]))
-            for f in files:
-                shutil.copy(os.path.join(root, f),
-                            os.path.join(directory_with_tests, root[n:], f))
+        source = Path(__file__).parent / 'testrunner-ex'
+        shutil.copytree(source, directory_with_tests)
 
         self.defaults = [
-            '--path', directory_with_tests,
+            '--path', str(directory_with_tests),
             '--tests-pattern', '^sampletestsf?$',
-            '--xml=.',
+            f'--xml={self.tmpdir}',
             ]
-        self._cleanup(self.reports_folder)
 
     def _run_tests(self):
         stream1 = io.StringIO()
@@ -166,7 +154,7 @@ class TextXMLOutputWithErrors(Base):
         self._run_tests()
         report = (
             self.reports_folder /
-            'doctest-src-zope-testrunner-tests-testrunner-ex-sample2-e.rst.xml'
+            'sample2.sampletests_f.Test.xml'
         )
         self.assertTrue(report.exists())
 
@@ -176,4 +164,5 @@ class TextXMLOutputWithErrors(Base):
         self.assertIn(' failures="1" ', content)
 
         # The failure is reported:
-        self.assertIn('Exception raised:', content)
+        self.assertIn("class 'AssertionError'", content)
+        self.assertIn("self.assertEqual(1, 0)", content)

